### PR TITLE
Add more flavor to deplyoments and pods

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @bbguimaraes

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
`pylint` a.k.a. the devil

**Comment by stevekuznetsov** [comment]
```
hack/monitor_prow.py:41:0: R0915: Too many statements (58/50) (too-many-statements)
```

Don't tell me what is too many, silly robot

**Comment by stevekuznetsov** [comment]
What does that even mean

File: hack/monitor_prow.py [hunk 1]
```diff
@@ -42,28 +48,92 @@ def highlight(log_dir, dc):
...
+        debug("deployment/{}: gathering info".format(dc))
+        header, deployError = renderHeader(dc)
         lines = []
+        log_lines = []
         for pod in exec_cmd(
                 'oc', 'get', 'pods', '--namespace', 'ci',
                 '--selector', 'component={}'.format(dc),
                 '--output', 'jsonpath={.items[*].metadata.name}').split():
+            debug("deployment/{}: pod/{}: gathering info".format(dc, pod))
+            lines.append(*renderFlavor(pod, dc))
```
Comment (bbguimaraes) [inline hack/monitor_prow.py]: `list.append` doesn't handle the zeroth case, we will get

`TypeError: append() takes exactly one argument (0 given)`

Comment (bbguimaraes) [reply hack/monitor_prow.py]:
```suggestion
            lines.extend(renderFlavor(pod, dc))
```

Comment (stevekuznetsov) [reply hack/monitor_prow.py]: wowza

**Comment by stevekuznetsov** [comment]
@bbguimaraes addressed

File: hack/monitor_prow.py [hunk 2]
```diff
@@ -42,28 +49,94 @@ def highlight(log_dir, dc):
...
+    spec = rawdc.get("spec", {})
+    status = rawdc.get("status", {})
+    desired = spec.get("replicas", 0)
+    current = status.get("replicas", 0)
+    updated = status.get("updatedReplicas", 0)
+    available = status.get("availableReplicas", 0)
+    version = "<unknown-version>"
+    containers = spec.get("template", {}).get("spec", {}).get("containers", [])
+    for container in containers:
+        if container.get("name", "") == dc:
```
Comment (petr-muller) [inline hack/monitor_prow.py]: `get()` has a `default=None` so you do not need to pass `""` as a second arg here

File: hack/monitor_prow.py [hunk 3]
```diff
@@ -42,28 +49,94 @@ def highlight(log_dir, dc):
...
             for l in exec_cmd(*cmd).splitlines():
                 if warn in l:
-                    lines.append(YELLOW + l + RESET)
+                    log_lines.append(YELLOW + l + RESET)
                 elif error in l or fatal in l:
-                    lines.append(RED + l + RESET)
+                    log_lines.append(RED + l + RESET)
+
+        if not log_lines and not lines:
+            header = header + " " + GREEN + "OK" + RESET
```
Comment (petr-muller) [inline hack/monitor_prow.py]: Pythonista in me prefers `"".join()` or `format()` or string interpolation here. Most of the code here uses `format`, I guess.

File: hack/monitor_prow.py [hunk 4]
```diff
@@ -42,28 +49,94 @@ def highlight(log_dir, dc):
...
+    desired = spec.get("replicas", 0)
+    current = status.get("replicas", 0)
+    updated = status.get("updatedReplicas", 0)
+    available = status.get("availableReplicas", 0)
+    version = "<unknown-version>"
+    containers = spec.get("template", {}).get("spec", {}).get("containers", [])
+    for container in containers:
+        if container.get("name", "") == dc:
+            image = container.get("image", "")
+            version = image[image.index(":")+1:]
```
Comment (petr-muller) [inline hack/monitor_prow.py]: `image.split(":", 1)[1]` might be slightly more readable

File: hack/monitor_prow.py [hunk 5]
```diff
@@ -42,28 +49,94 @@ def highlight(log_dir, dc):
...
+
+
+def renderFlavor(pod, dc):
+    debug("deployment/{}: pod/{}: rendering flavor".format(dc, pod))
+    lines = []
+    raw = json.loads(exec_cmd('oc', 'get', 'pod/{}'.format(pod),
+                              '--namespace', 'ci', '--output', 'json'))
+    for container in raw.get("status", {}).get("containerStatuses", []):
+        debug("pod/{}: handling status for container {}".format(pod, container.get("name", "")))
+        if container.get("name", "") == dc:
```
Comment (petr-muller) [inline hack/monitor_prow.py]: The `default=None` thing again, plus in the rest of this method. Especially using `""` as the not-present indicator is suspicious, this should be either `None`, or the matching-type empty value of what `get` is expected to return (usually dictionaries here).

File: hack/monitor_prow.py [hunk 6]
```diff
@@ -42,28 +49,94 @@ def highlight(log_dir, dc):
...
+            state = container.get("state", {})
+            if state.get("running", "") == "":
+                waiting = state.get("waiting", "")
+                if waiting != "":
```
Comment (petr-muller) [inline hack/monitor_prow.py]: Any reason to not use `not waiting`?

**Pull request merged**